src/ucp/core/ucp_mm.c

```diff
@@ -322,7 +302,16 @@ void ucp_memh_unmap(ucp_context_h context, ucp_mem_h memh, ucp_md_map_t md_map)
         md_map  &= ~UCS_BIT(memh->alloc_md_index);
     }
 
-    ucp_memh_dereg(context, memh, md_map);
+    if ((memh->parent != NULL) && (memh->parent != memh)) {
+        /* TODO remove invalidate, replace with ucp_memh_dereg weak MDs */
+        if (context->rcache != NULL) {
```

Comment:
isn't the check in line 305 already implies rcache is on (since we have "real" parent)?
so can convert to assert